src/main/java/edu/harvard/iq/dataverse/EjbDataverseEngine.java

```diff
@@ -215,7 +218,21 @@ private DvObject getRetType(Object r){
             }
 
             DataverseRequest dvReq = aCommand.getRequest();
-            
+
+            AuthenticatedUser authenticatedUser = dvReq.getAuthenticatedUser();
+            if (authenticatedUser != null) {
+                AuthenticatedUser auFreshLookup = authentication.findByID(authenticatedUser.getId());
+                if (auFreshLookup == null) {
+                    logger.fine("submit method found user no longer exists (was deleted).");
+                    throw new CommandException(BundleUtil.getStringFromBundle("command.exception.user.deleted", Arrays.asList(aCommand.getClass().getSimpleName())), aCommand);
+                } else {
+                    if (auFreshLookup.isDeactivated()) {
+                        logger.fine("submit method found user is deactivated.");
+                        throw new CommandException(BundleUtil.getStringFromBundle("command.exception.user.deactivated", Arrays.asList(aCommand.getClass().getSimpleName())), aCommand);
+                    }
+                }
+            }
+
```

Comment:
I think we can simplify and remove this - there is already a DataverseRequest associated with this call to the command and that should be enough, as long as we also change the one place where it gets the user from the session in DataverseRequestService to call the new getUser(true) verson of that method.
```
   @PostConstruct
    protected void setup() {
        dataverseRequest = new DataverseRequest(dataverseSessionSvc.getUser(), request);
    }
```

Reply:
I tried putting it there in the postconstruct and couldn't get it to work. So I put it in the engine instead.

Reply:
I don't remember a error. I think it simply didn't work. It had no effect, as I recall.

Reply:
That's weird - since it should set the request to point to a GuestUser in that case and therefore fail due to permission. I would prefer this simpler mechanism if we can get it to work, though (and at least understand why it doesn't if it doesn't). Let me go ahead and try it and see what happens.

Reply:
Ok. Good luck.